server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -240,11 +241,25 @@ public IndexAbstraction resolveWriteIndexAbstraction(ClusterState state, DocWrit
         }
     }
 
-    private static Collection<String> resolveExpressions(Context context, String... expressions) {
-        return WildcardExpressionResolver.resolve(
-            context,
-            DateMathExpressionResolver.resolve(context, expressions == null ? List.of() : List.of(expressions))
-        );
+    protected static Collection<String> resolveExpressions(Context context, String... expressions) {
```

Comment:
Compared to the original method, the refactored one groups all the special cases and deals with them earlier in this single place, rather than handle them while resolving wildcards of regular exceptions.